### PR TITLE
HOTFIX #2066: Revert changes to versioning information

### PR DESCRIPTION
**Pull request by astrogeco**
**Describe the contribution**

Per conversation in CCB:2022-03-24 and in https://github.com/nasa/cFE/commit/e5be0612ad6bf9db09ddf640306a65292abce2d5#commitcomment-69358417

Ensuring REVISION number is set to 99
Correct cfs_versions.dox appropriately. 

**Testing performed**
Text and documentation only

**Expected behavior changes**
No behavior change

**System(s) tested on**
n/a

**Additional context**
https://github.com/nasa/cFE/commit/e5be0612ad6bf9db09ddf640306a65292abce2d5#commitcomment-69358417

**Third party code**
none

**Contributor Info - All information REQUIRED for consideration of pull request**
Gerardo E. Cruz-Ortiz, NASA

### CODE REVIEW

File: modules/core_api/fsw/inc/cfe_version.h [hunk 1]
```diff
@@ -28,21 +28,20 @@
...
-#define CFE_BUILD_NUMBER   80           /**< @brief Development: Number of development commits since baseline */
+#define CFE_BUILD_NUMBER   80 /**< @brief Development: Number of development git commits since CFE_BUILD_BASELINE */
 #define CFE_BUILD_BASELINE "v7.0.0-rc4" /**< @brief Development: Reference git tag for build number */
 
-/* Version Macro Definitions updated for official releases only */
-#define CFE_MAJOR_VERSION 6 /**< @brief Major release version (Former for Revision == 99) */
-#define CFE_MINOR_VERSION 7 /**< @brief Minor release version (Former for Revision == 99) */
-#define CFE_REVISION \
-    99 /*!< @brief * Set to 0 on OFFICIAL releases, and set to 99 on development versions. Revision number.      */
+/* See cfs_versions.dox for definitions */
```
Comment (astrogeco) [inline modules/core_api/fsw/inc/cfe_version.h:34]: Is there a doxygen "link command" of sorts to turn this into a hyperlink in the generated documentation?

Comment (skliper) [reply modules/core_api/fsw/inc/cfe_version.h:34]: yes, although you need to link to the page name... see https://github.com/nasa/CF/blob/d0b3e25566b7cb8221d8a3f90017fe52ee0f43d4/fsw/src/cf_version.h#L31

File: modules/core_api/fsw/inc/cfe_version.h [hunk 2]
```diff
@@ -28,21 +28,20 @@
...
+#define CFE_REVISION      99
 
 /*!
  * @brief Mission revision.
  *
- * Reserved for mission use to denote patches/customizations as needed.
+ * Reserved for mission use to denote patches/customizations as needed. 
  */
-#define CFE_MISSION_REV 0
+#define CFE_MISSION_REV 0xFF
```
Comment (astrogeco) [inline modules/core_api/fsw/inc/cfe_version.h:45]: Are we good with this or should we add something like 0-244 being "reserved" for missions but 0xFF *may* indicate development

File: modules/core_api/fsw/inc/cfe_version.h [hunk 3]
```diff
@@ -28,21 +28,20 @@
...
 
-/* Version Macro Definitions updated for official releases only */
-#define CFE_MAJOR_VERSION 6 /**< @brief Major release version (Former for Revision == 99) */
-#define CFE_MINOR_VERSION 7 /**< @brief Minor release version (Former for Revision == 99) */
-#define CFE_REVISION \
-    99 /*!< @brief * Set to 0 on OFFICIAL releases, and set to 99 on development versions. Revision number.      */
+/* See \ref cfsversions for definitions */
+#define CFE_MAJOR_VERSION 6
+#define CFE_MINOR_VERSION 7
+#define CFE_REVISION      99
```
Comment (skliper) [inline modules/core_api/fsw/inc/cfe_version.h:37]: Can you add the doxygen comment like https://github.com/nasa/CF/blob/d0b3e25566b7cb8221d8a3f90017fe52ee0f43d4/fsw/src/cf_version.h#L35-L37

Comment (astrogeco) [reply modules/core_api/fsw/inc/cfe_version.h]: done

File: modules/core_api/fsw/inc/cfe_version.h (hunk 2 above)
Comment (skliper) [reply modules/core_api/fsw/inc/cfe_version.h:45]: 1-254 you mean?  0 and 255 we've claimed.  You could just reference the issue?  Not sure how helpful that is though.

File: modules/core_api/fsw/inc/cfe_version.h [hunk 4]
```diff
@@ -28,21 +28,20 @@
...
+#define CFE_MINOR_VERSION 7  /**< @brief Minor version number */
+#define CFE_REVISION      99 /**< @brief Revision version number */
 
 /*!
  * @brief Mission revision.
  *
  * Reserved for mission use to denote patches/customizations as needed.
  */
-#define CFE_MISSION_REV 0
+#define CFE_MISSION_REV 0xFF
```
Comment (astrogeco) [inline modules/core_api/fsw/inc/cfe_version.h:45]: @skliper Are we good with this or should we add something like 0-244 being "reserved" for missions but 0xFF may indicate development

Comment (skliper) [reply modules/core_api/fsw/inc/cfe_version.h:45]: maybe just add "NOTE - reserving 0 and 0xFF currently for open source use, although pending resolution of https://github.com/nasa/cFS/issues/440"

Comment (astrogeco) [reply modules/core_api/fsw/inc/cfe_version.h:45]: I like that

**Pull request merged**